toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -72,43 +92,508 @@ def send_dm_to_user(
+    is a member of.
+    """
+    if isinstance(conversation_types, ConversationType):
+        conversation_types = [conversation_types]
+
+    conversation_types_filter = ",".join(
+        convert_conversation_type_to_slack_name(conv_type).value
+        for conv_type in conversation_types or ConversationType
+    )
+
+    slackClient = AsyncWebClient(token=context.authorization.token)
+
+    results, next_cursor = await async_paginate(
+        slackClient.conversations_list,
+        "channels",
+        limit=limit,
+        next_cursor=next_cursor,
+        types=conversation_types_filter,
+        exclude_archived=True,
+    )
+
+    return {
+        "conversations": [
+            extract_conversation_metadata(conversation)
+            for conversation in results
+            if conversation.get("is_im") or conversation.get("is_member")
+        ],
+        "next_cursor": next_cursor,
+    }
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:read"],
+    )
+)
+async def list_public_channels_metadata(
+    context: ToolContext,
+    limit: Annotated[
+        Optional[int], "The maximum number of channels to list."
+    ] = MAX_PAGINATION_SIZE_LIMIT,
+) -> Annotated[dict, "The public channels"]:
+    """List metadata for public channels in Slack that the user is a member of."""
+
+    return await list_conversations_metadata(
+        context,
+        conversation_types=[ConversationType.PUBLIC_CHANNEL],
+        limit=limit,
+    )
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["groups:read"],
+    )
+)
+async def list_private_channels_metadata(
+    context: ToolContext,
+    limit: Annotated[
+        Optional[int], "The maximum number of channels to list."
+    ] = MAX_PAGINATION_SIZE_LIMIT,
+) -> Annotated[dict, "The private channels"]:
+    """List metadata for private channels in Slack that the user is a member of."""
+
+    return await list_conversations_metadata(
+        context,
+        conversation_types=[ConversationType.PRIVATE_CHANNEL],
+        limit=limit,
+    )
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["mpim:read"],
+    )
+)
+async def list_group_direct_message_channels_metadata(
+    context: ToolContext,
+    limit: Annotated[
+        Optional[int], "The maximum number of channels to list."
+    ] = MAX_PAGINATION_SIZE_LIMIT,
+) -> Annotated[dict, "The group direct message channels"]:
+    """List metadata for group direct message channels in Slack that the user is a member of."""
+
+    return await list_conversations_metadata(
+        context,
+        conversation_types=[ConversationType.MULTI_PERSON_DIRECT_MESSAGE],
+        limit=limit,
+    )
+
+
+# Note: Bots are included in the results.
+# Note: Direct messages with no conversation history are included in the results.
+@tool(
+    requires_auth=Slack(
+        scopes=["im:read"],
+    )
+)
+async def list_direct_message_channels_metadata(
```

Comment:
Example response: `{"conversations":[{"conversation_type":"im","id":"D088LB2QP7T","is_archived":false,"is_member":true,"is_private":true,"is_user_deleted":false,"name":null,"num_members":2,"purpose":"","user":"U087T4BC415"}}`

Question: would it be reasonable to get the username and other info for each person I have a dm with? Would this be too expensive?